git_hooks/pre-push

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+# Called by "git push" after it has checked the remote status, but before anything has been
+# pushed.  If this script exits with a non-zero status nothing will be pushed.
+#
+# This hook is called with the following parameters:
+#
+# $1 -- Name of the remote to which the push is being done
+# $2 -- URL to which the push is being done
+#
+# If pushing without using a named remote those arguments will be equal.
+#
+# Information about the commits which are being pushed is supplied as lines to
+# the standard input in the form:
+#
+#   <local ref> <local sha1> <remote ref> <remote sha1>
+#
+
+remote="$1"
+url="$2"
+
+echo "remote is $remote"
+echo "url is $url"
+
+./tests/scripts/check-doxy-blocks.pl
+if [ $? != 0 ]
+then
+exit 1
+fi
+
+./tests/scripts/check-names.sh
+if [ $? != 0 ]
+then
+exit 1
+fi
+
+./tests/scripts/check-generated-files.sh
+if [ $? != 0 ]
+then
+exit 1
+fi
```

Comment:
This seems to be a recurrent pattern i.e. a test command followed by a if statement to check whether the execution was successful. Perhaps we should use a shell "function" to avoid code duplication and to make the script easily extensible in the future. I suggest something like the following:
```
check()
{
    # Run command in $1 argument and check return code/output
}

check "./tests/scripts/check-doxy-blocks.pl"
check "./tests/scripts/check-names.sh"
check "./tests/scripts/check-generated-files.sh"
```

Please check [here](https://github.com/andresag01/mbedtls/blob/all-sh-armc6/scripts/output_env.sh#L20) for an example of what I mean.